R/enrichment_map.R

```diff
@@ -185,6 +221,57 @@ enrichment_map <- function(res_enrich,
   # re-sorting the vertices alphabetically
   rank_gs <- rank(V(emg)$name)
   emg <- permute.vertices(emg, rank_gs)
+  
+  # add communities based on different community detection algorithms
+  add_communities <- function(em) {
+    cd_algorithms <- c(cluster_louvain, 
+                       cluster_edge_betweenness, 
+                       cluster_fast_greedy)
+    names(cd_algorithms) <- c("louvain", 
+                              "in_betweenness", 
+                              "fast_greedy")
+    i <- 1
+    for (algo in cd_algorithms) {
+      cl <- algo(em)$membership
+      col_background <- map2color(cl, mypal, range(cl))
+      col_highlight <- map2color(cl, mypal_select, range(cl))
+      col_hover <- map2color(cl, mypal_hover, range(cl))
+      em <- set_vertex_attr(graph = em, 
+                            name = names(cd_algorithms)[i], 
+                            value = cl)
+      em <- set_vertex_attr(graph = em, 
+                            name = paste0(names(cd_algorithms)[i], "_background"), 
+                            value = col_background)
+      em <- set_vertex_attr(graph = em, 
+                            name = paste0(names(cd_algorithms)[i], "_highlight"), 
+                            value = col_highlight)
+      em <- set_vertex_attr(graph = em, 
+                            name = paste0(names(cd_algorithms)[i], "_hover"), 
+                            value = col_hover)
+      i <- i + 1
+    }
+    return(em)
+  }
+  
+  emg <- add_communities(emg)
 
   return(emg)
 }
+
+create_interactive_map <- function(em, color_by = "gs_pvalue") {
+  idx <- grep(color_by, x = vertex_attr_names(em))
+  if(length(idx) > 0) {
+    V(em)$color.background <- vertex_attr(em, paste0(color_by, "_background"))
+    V(em)$color.highlight <- vertex_attr(em, paste0(color_by, "_highlight"))
+    V(em)$color.hover <- vertex_attr(em, paste0(color_by, "_hover"))
+  }
+  vis_graph <- visIgraph(em)
+  select_options <- vertex_attr_names(em)[grep("_background", x = vertex_attr_names(em))]
+  select_options <- gsub("_background", "", select_options)
+  vis_graph <- visOptions(vis_graph, highlightNearest = list(enabled = TRUE, 
+                                                             degree = 1, 
+                                                             hover = TRUE),
+                          nodesIdSelection = TRUE,
+                          selectedBy = list(variable="louvain", variable="in_betweenness"))
```

Comment:
Is the list accepting two arguments of the same name? I could not display the second one